crates/client/rpc/src/lib.rs

```diff
@@ -137,11 +137,10 @@ where
     pub fn current_block_hash(&self) -> Result<H256, ApiError> {
         let substrate_block_hash = self.client.info().best_hash;
 
-        let starknet_block =
-            get_block_by_block_hash(self.client.as_ref(), substrate_block_hash).ok_or(ApiError::UnknownBlock(
-                format!("Failed to get Starknet block for Substrate block hash '{substrate_block_hash}' "),
-            ))?;
-
+        let starknet_block = match get_block_by_block_hash(self.client.as_ref(), substrate_block_hash) {
+            Ok(block) => block,
+            Err(err) => return Err(ApiError::UnknownBlock(err.to_string())),
```

Comment:
StarknetRpcApiError::BlockNotFound

Reply:
same maybe just use `?`

Reply:
@tdelabro 
```rust
137 pub fn current_block_hash(&self) -> Result<H256, ApiError> 
```
Calling function `current_block_hash` signature requires ApiError, do you mean I change `current_block_hash` signature to return StarknetRpcApiError? Thanks

Reply:
yes